File: frontend/src/talkog.ts
export type Talkog = {
  userId: string;
  talk: string;
  timestamp: Date;
};

export const create = (
  userId: string,
  talk: string,
  timestamp: number
): Talkog => {
  return {
    userId,
    talk,
    timestamp: new Date(timestamp * 1000),
  };
};

export const equals = (t1: Talkog, t2: Talkog) => {
  return (
    t1.userId === t2.userId &&
    t1.talk === t2.talk &&
    t1.timestamp.getTime() === t2.timestamp.getTime()
  );
};

File: backend/talkog/talkog.go
package talkog

import "time"

type Talkog struct {
	UserId    string // UUID
	Timestamp int64  // Unix Time
	Talk      string
}

func New(userId string, talk string) Talkog {
	return Talkog{
		UserId:    userId,
		Timestamp: time.Now().Unix(),
		Talk:      talk,
	}
}
